server/devdocs/selectors/function-comments.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Find the node containing the docblock
+ * for a function given its name
+ *
+ * JSDoc function comments are attached to AST nodes
+ * in a JavaScript file but these aren't always with
+ * the direct nodes defining the functions.
+ *
+ * For example, if a function is exported such as
+ * below we end up with this comment being associated
+ * with the export declaration and the function itself
+ * is a child of the export declaration node.
+ *
+ * Therefore we need to find the "outermost" node onto
+ * which an associated docblock could be attached for
+ * a function of the given name.
+ *
+ * @see http://eslint.org/parser/ or similar tool for an interactive AST explorer
+ *
+ * Find these types of occurrences:
+ *  - `export function noop() {}` -> Return the export declaration
+ *  - `export const noop = function() {}` -> Return the export declaration
+ *  - `export const noop = () => {}` -> Return the export declaration
+ *  - `function noop() {}` -> Return the function declaration
+ *  - `const noop = function() {}` -> Return the variable declaration
+ *  - `const noop = () => {}` -> Return the variable declaration
+ *
+ * @param {Object} ast an espree-generated AST
+ * @param {String} name name of function for associated docblock
+ * @returns {?Object} AST node containing block comment if exists
+ */
+export const findOutermostNode = ( ast, name ) =>
+	ast && ast.body && ast.body.find( ( { declaration, declarations, id, type } ) => (
+		(
+			type === 'ExportNamedDeclaration' &&
+			(
+				(
+					declaration.type === 'FunctionDeclaration' &&
+					declaration.id.name === name
+				) ||
+				(
+					declaration.type === 'VariableDeclaration' &&
+					declaration.declarations &&
+					declaration.declarations[ 0 ] &&
```

Comment:
Could it ever be the case that a type is `VariableDeclaration` and `declarations` or `declarations[ 0 ]` be falsey?